routers/web/repo/release.go

```diff
@@ -214,6 +214,8 @@ func TagsList(ctx *context.Context) {
 	ctx.Data["HideBranchesInDropdown"] = true
 	ctx.Data["CanCreateRelease"] = ctx.Repo.CanWrite(unit.TypeReleases) && !ctx.Repo.Repository.IsArchived
 
+	namePattern := ctx.FormTrim("q")
```

Comment:
I'm not really a fan of allowing end users to set SQL LIKE statements (i.e. they might not expect that entering `v1_%` shows all tags in Version 1X).
However, what would be the alternative?
Parsing a glob instead and mapping it to the LIKE syntax?

Reply:
If we choose to merge this as it is, please add a docs page in https://gitea.com/gitea/docs describing that this search accepts the SQL LIKE syntax.

Reply:
The ideia behind is to macht all tags whose related commit message contains the criterion (i.e., if the user passes "2" as the criterion. It should matches v2, v1.2, v.1.32, and so on). I double-checked the generated SQL and xorm puts the wildcard on both sides of the literal (see evidence attached). This behavior is inline with how the commit search works.

![image](https://github.com/user-attachments/assets/321296d7-126e-4abc-9a50-a5c1ee709598)

About the documentation changes, gonna submitt a PR latter today :)

Reply:
It uses `builder.Like{"lower_tag_name", strings.ToLower(opts.NamePattern.Value())}`, which will generate a SQL like `lower_tag_name = ?`, And the `%` will be added automatically if it hasn't been added.

Reply:
@delvh do you think we still need to add a docs page for this particular change after the explanation ? I can't seem to find any existing page that is closely related to this change (the closest seem to be `Usage > Protected tags` but I feel is not quite right). Any ideas ?

Reply:
My problem is simply that I don't like hiding functionality from the user in intransparent ways.
What we could do as a simple workaround is add a tooltip to the search bar stating
`Search for matching tags. Use '%' as a placeholder for any sequence of chars and  '_' for a single unknown char` or something like that.

Reply:
@delvh Cool, I'm gonna add the tooltip latter today :)

Reply:
@delvh What do you think ?

![image](https://github.com/user-attachments/assets/757bddc1-8782-41ec-8638-b22ffd211795)

If you guys think it is ok, I'll push the changes :)

Reply:
Last thing: I'm not quite happy with using the query param `q` as tag name.
I mean… it is sort of a query.
However, it might be confusing in the future if there are multiple search boxes why one is `q` while the other one isn't.

Reply:
This one could be a little tricky. The thing is that the page is using the [combo.tmpl](https://github.com/go-gitea/gitea/blob/main/templates/shared/search/combo.tmpl) template. Which, in turn uses the [input,tmpl](https://github.com/go-gitea/gitea/blob/main/templates/shared/search/input.tmpl). The latter (source-code attached) assumes the name of the field to be `q`. I could change it to receive a custom name, but it feels that it is out of the scope of this particular PR.

How do you guys want to proceed ? Do you want me to create another PR that allows a custom name for the combo component and them rebase the this one when the new one is merged ? Or do you want me to put it all together on a single PR ?

![image](https://github.com/user-attachments/assets/32182887-18bd-4c82-bd02-9e79543be4c0)